### ISSUE TITLE
Improvment to 97% accuracy

### ISSUE THREAD

**Issue by Ananas120** [oscarknagg/one-shot-speaker-identification#10]
Hello, i don’t know if this repo is active but if i can help, i used this repo for my project and i find a method to improve the loss / accuracy just by L2-normalizing the output of the encoder 
My score with it is actually 0.97% accuracy and 0.02 val-BCEloss training for 25 epochs on a mixt of LibriSpeech and CommonVoice (fr) datasets (360 speakers in train set and 150 in validation set with 200 pairs for each speaker (100 same and 100 not same) (batch_size of size 32 (16 same and 16 not) with embedding dim 64)

**Comment by Vatsalparsaniya** [comment]
@Ananas120  Hi, are you using python 2.7 or 3.*?

**Comment by Ananas120** [comment]
I re-implemented your model architecture in python 3 in my personal project

**Comment by Vatsalparsaniya** [comment]
Thank you for your response,
I have just reimplemented train_siamese.py code in python 3 and currently training on google colab.
![image](https://user-images.githubusercontent.com/33985480/117288827-8af50c80-ae89-11eb-944f-7f17a51161da.png)

I'm getting accuracy value  ==  0.5000 since first training step. Is it normal or I'm missing something.
if it is possible can you share training log?

Thank you.

**Comment by Ananas120** [comment]
Yes it is normal ! I found that siamese networks need some epoch (sometimes more than 5 or 10 !) to increase their accuracy and this can be explained as follow : they try to separate « same » and « not same » from a scalar value (distance) so they have to find the right threshold before having a better score (before that they always predict either 0 or 1)
